test/e2e-go/e2e_go_cluster_test.go

```diff
@@ -177,13 +185,9 @@ var _ = Describe("operator-sdk", func() {
 			}
 			Eventually(getCurlLogs, time.Minute, time.Second).Should(ContainSubstring("< HTTP/2 200"))
 
-			By("cleaning up the curl pod")
-			_, err = tc.Kubectl.Delete(true, "pods", "curl")
-			Expect(err).NotTo(HaveOccurred())
-
 			By("validating that the created resource object gets reconciled in the controller")
 			Expect(metricsOutput).To(ContainSubstring(fmt.Sprintf(
-				`controller_runtime_reconcile_total{controller="%s",result="success"} 1`,
+				`controller_runtime_reconcile_total{controller="%s",result="success"} 3`,
 				strings.ToLower(tc.Kind),
```

Comment:
This change is brittle for a few reasons, chiefly that it depends on what order the test specs are run (this is the 3rd run out of 4). If another test spec is added to the golang suite and it gets run before the cluster test, or the cluster test runs by itself, this line will break. You can fix this by incrementing a global `count` variable in `BeforeEach` and using that variable here.
```suggestion
				`controller_runtime_reconcile_total{controller="%s",result="success"} %d`,
				strings.ToLower(tc.Kind), specCount,
```

Reply:
The quantity in this scenario is specified via code (webohook): 

```
// Default implements webhook.Defaulter so a webhook will be registered for the type
func (r *Memcached) Default() {
	memcachedlog.Info("default", "name", r.Name)

	if r.Spec.Size == 0 {
		r.Spec.Size = 3
	}
}
```
So, in this case, shows that the best approach is to keep it fixed.

Reply:
That is incorrect. This metric collects the number of reconciles that have occurred. It used to be 1 because each project had a different name/GVK. Now that memcached-operator/Memcached is used for all test project names, the reconcile count == number of tests that trigger a reconcile.

Reply:
>  This metric collects the number of reconciles that have occurred. It used to be 1 because each project had a different name/GVK 👍 

> Now that memcached-operator/Memcached is used for all test project names, the reconcile count == number of tests that trigger a reconcile. 👎

It is true only for the current golang e2e test code implementation. In a test, we can do more than on reconciliation. See the helm one for example. 
 
> This change is brittle for a few reasons, chiefly that it depends on what order the test specs are run (this is the 3rd run out of 4). If another test spec is added to the golang suite and it gets run before the cluster test, or the cluster test runs by itself, this line will break. You can fix this by incrementing a global count variable in BeforeEach and using that variable here. 👎 

The TestContext object represents the context for ALL tests. Then, 
- Would an attrs reconcile_count make sense for ANY possible test ? No. So, it shows not be part of the TestContext domain. We ought to not add new attrs to the structure to solve specific scenarios for specific tests. 
- Could we assume that `by incrementing a global count variable in BeforeEach and using that variable` it will have the expected result always no matter the changes made in the tests? No. As described above we can trigger that more than once.

IMO: 
- (option A) We can keep the fixed number. If we change the tests and it changes the number of reconciles count then, we ought to update the here the fixed number as we did now ( replace 1 per 3 )
OR
- (option B) We should keep the number and let this check only to ensure that the `controller_runtime_reconcile_total` metrics will be found for the controller pod manager with the result as a success. 
OR
- (option C) Remove the check if you think that it is no valid. 

Based on the scope of this pr ( replace the e2e mock by the samples) my vote here is we move forward with the option A.

Reply:
> So, it shows not be part of the TestContext domain

I said create a global variable not add a field to TestContext.

> it will have the expected result always no matter the changes made in the tests

The current situation is worse, since the following is broken (running only one test within the suite):

```sh
go test ./test/e2e-go/... -ginkgo.focus=.*operator-sdk.*
```

However the test setup doesn’t encourage doing this so we could be ok hard coding the count for this reason alone.

> It is true only for the current golang e2e test code implementation. In a test, we can have > 1.

So increment the count variable every time you CRUD the CR in the test function. The operator is incredibly simple so the CR won’t actually change unless you `kubectl` it.

Another way to know if a CR was reconciled is to update a `status.reconcileCount` field in the controller, which we may want to consider since using a sample for multiple tests in the same cluster causes some unexpected conflicts like this.